### ISSUE TITLE
Weird growth of KnowledgeCommunity size to 172 peers

### ISSUE THREAD

**Issue by synctext** [Tribler/tribler#8292]
Yesterday and today I was able to reproduce the following weirdness on latest V8.0 on Ubuntu 20.04 LTS. Use number of peers in 1 trouble community.
![Screenshot from 2024-11-30 20-24-32](https://github.com/user-attachments/assets/11243e40-d2ff-44f2-b940-bb3e3abffa90)
After a restart of Tribler you can see it slowly climbing (slow .GIF of 70 seconds):
![Knowledge_community_growth__Nov2024](https://github.com/user-attachments/assets/38228e63-d33b-4629-b654-1dd0d0e8c15b)
Very weird, good luck with this one!

**Comment by synctext** [comment]
After a night of running Tribler the yield is 291 peers :fearful: : 
![Screenshot from 2024-12-01 08-38-43](https://github.com/user-attachments/assets/f4e3b26c-c32c-4650-993e-b3f326c9eb79)

**Comment by qstokkink** [comment]
The _theoretical maximum_ number of peers that should appear in the KnowledgeCommunity under normal IPv8 operation is 216. So, 291 peers is definitely not normal.

So, either there is a bug in IPv8 or the KnowledgeCommunity is misbehaving.

**Comment by qstokkink** [comment]
First, I verified that IPv8's internal peer walking/dropping was correctly working. There were no errors.

Second, I checked that the `KnowledgeCommunity` was not receiving messages from unknown peers. I added a check to `on_message()` and `on_request()`. The former had no issues. However, the latter (`on_request()`) receives a lot of messages from unknown peers:

```python
src/tribler/core/knowledge/community.py", line 107, in on_request
    assert peer in self.get_peers()
AssertionError
```
